pkg/installmanager/installmanager.go

```diff
@@ -619,59 +664,54 @@ func (m *InstallManager) loadClusterDeployment() (*hivev1.ClusterDeployment, err
 	return cd, nil
 }
 
-func gatherLogs(cd *hivev1.ClusterDeployment, m *InstallManager) error {
-	m.log.Info("Gathering logs/tarball")
+func (m *InstallManager) gatherLogs(cd *hivev1.ClusterDeployment) (string, error) {
+	m.log.Info("gathering logs from cluster")
 
-	m.log.Debug("Checking for SSH private key")
+	m.log.Debug("checking for SSH private key")
 	sshPrivKeyPath := os.Getenv("SSH_PRIV_KEY_PATH")
 	if sshPrivKeyPath == "" {
 		m.log.Warn("cannot gather logs as SSH_PRIV_KEY_PATH is unset or empty")
-		return nil
+		return "", fmt.Errorf("cannot gather logs as SSH_PRIV_KEY_PATH is unset or empty")
 	}
 	fileInfo, err := os.Stat(sshPrivKeyPath)
 	if err != nil && os.IsNotExist(err) {
 		m.log.Warn("cannot gather logs/tarball as no ssh private key file found")
-		return err
+		return "", err
 	} else if err != nil {
 		m.log.WithError(err).Error("error stating file containing private key")
-		return err
+		return "", err
 	}
 
 	if fileInfo.Size() == 0 {
 		m.log.Warn("cannot gather logs/tarball as ssh private key file is empty")
-		return err
+		return "", err
 	}
 
 	// set up ssh private key, and run the log gathering script
 	cleanup, err := initSSHAgent(sshPrivKeyPath, m)
 	defer cleanup()
 	if err != nil {
 		m.log.WithError(err).Error("failed to setup SSH agent")
-		return err
+		return "", err
 	}
 
 	bootstrapIP, err := getBootstrapIP(m)
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	logTarball, err := runGatherScript(bootstrapIP, fetchLogsScriptTemplate, m)
+	logTarball, err := m.runGatherScript(bootstrapIP, fetchLogsScriptTemplate, m.WorkDir)
 	if err != nil {
 		m.log.Error("failed to run script for gathering logs")
-		return err
+		m.log.Warn("sleeping for a long long time")
```

Comment:
Did you intend to leave this in here? Or is this a remnant from your testing?

Reply:
Remnant of testing.